js/map.js

```diff
@@ -27,13 +27,17 @@
 
   var onPinClick = function (pin, advert) {
     deactivatePins();
+
+    window.card.remove();
+    window.card.show(advert, map);
     pin.classList.add('map__pin--active');
+  };
 
-    window.card.removeCard();
-    window.card.showCard(advert, map);
+  var handleOnCardRemove = function () {
```

Comment:
```suggestion
  var onCardRemove = function () {
```
Или добавь как анонимную функцию:

```js
window.card.setOnRemove(function () { 
  deactivatePins();
});
```